### PR TITLE
APR hokusai deployment

### PR DESCRIPTION
**Pull request by izakp**
- Configure hokusai templates for staging and production stacks

For staging:
- run `hokusai build && hokusai push staging` to build and create the `staging` image tag
- run `hokusai stack up staging`

For production:
- Delete the `apr` service and deployment in kubernetes production
- run `hokusai build && hokusai push production` to build and create the `production` image tag
- run `hokusai stack up production`
- switch DNS for `apr.artsy.net` to the new apr service ELB (get the DNS name with `kubectl describe service apr`)
- `cd terraform && terraform apply` to delete the old ELB

### CODE REVIEW

File: hokusai/production.yml [hunk 1]
```diff
@@ -31,13 +47,14 @@ metadata:
     app: apr
   name: apr
   namespace: default
+  annotations:
+    service.beta.kubernetes.io/aws-load-balancer-proxy-protocol: "*"
```
Comment (izakp) [inline hokusai/production.yml:56]: This *should* enable proxy protocol on the ELB - best to test on staging though

File: hokusai/staging-config.yml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+apiVersion: v1
```
Comment (izakp) [inline hokusai/staging-config.yml:1]: These were already pushed to the clusters with `hokusai config push staging` / `hokusai config push production`

Likewise I created secrets with `hokusai secrets push staging` / `hokusai secrets push production` - they are not checked in but you can pull them locally with  `hokusai secrets pull staging` / `hokusai secrets pull production`, edit them and push them again

File: circle.yml [hunk 3]
```diff
@@ -1,17 +1,28 @@
...
   override:
-    - docker build --build-arg BASIC_AUTH_USER=$BASIC_AUTH_USER --build-arg BASIC_AUTH_PASSWORD=$BASIC_AUTH_PASSWORD -t $AWS_ACCOUNT_ID.dkr.ecr.us-east-1.amazonaws.com/apr:$CIRCLE_SHA1 .
+    - sudo pip install docker-compose==1.9.0
+    - sudo apt-get install python-dev
+    - sudo pip install git+http://github.com/artsy/hokusai.git@master#egg=Hokusai
 
 test:
   override:
-    - echo "bypassing test for now"
+    - hokusai test
```
Comment (ashkan18) [inline circle.yml]: we may want to still bypass build here, APR doesn't have any test 😨

File: circle.yml [hunk 4]
```diff
@@ -1,17 +1,28 @@
...
 
 deployment:
-  prod:
+  staging:
     branch: master
     commands:
-      - ./deploy.sh
\ No newline at end of file
+      - curl -O https://storage.googleapis.com/kubernetes-release/release/v1.4.7/bin/linux/amd64/kubectl
```
Comment (ashkan18) [inline circle.yml]: I wonder if it make sense to have these next few generic lines in `.sh` and run that instead

**Review by ashkan18 — COMMENTED** [review]
Two #minor comments, this looks aweeeesome!

File: circle.yml (hunk 3 above)
Comment (izakp) [reply circle.yml]: Ah right

File: circle.yml (hunk 4 above)
Comment (izakp) [reply circle.yml]: Ok i'll put the kubectl configuration in a .sh file

Comment (izakp) [reply circle.yml]: Also did so for the hokusai install

**Comment by izakp** [comment]
@ashkan18 let's roll this out tomorrow AM

**Comment by izakp** [comment]
http://apr-staging.artsy.net/ is up!

**Pull request merged**